src/Features/CSharp/Portable/Completion/KeywordRecommenders/IntoKeywordRecommender.cs

```diff
@@ -40,6 +40,17 @@ private bool IsValidContextForSelect(CSharpSyntaxContext context)
 
             var lastToken = select.Expression.GetLastToken(includeSkipped: true);
 
+            var memberAccessExpression = select.Expression as MemberAccessExpressionSyntax;
```

Comment:
Maybe:
* check to see if `token` is a `DotToken`
* check to see if the `DotToken` is the child of a `MemberAccessExpresion` with a missing right node
* check if that `MemberAccessExpression` (or one of its ancestors) is `select.Expression`
?